core/ustring.cpp

```diff
@@ -2960,12 +2960,11 @@ String String::replace_first(const String &p_key, const String &p_with) const {
 	int search_from = 0;
 	int result = 0;
 
-	while ((result = find(p_key, search_from)) >= 0) {
+	if ((result = find(p_key, search_from)) >= 0) {
```

Comment:
This could likely be simplified further, it seems like this `break;` was hacked into a copy paste of `replace` as a lazy way to have it replace only one occurrence. But the `search_from` variable is unnecessary now.

Reply:
I'm not sure how this should exactly look.